ts/main.ts

```diff
@@ -0,0 +1,64 @@
+// interface for Monster
+interface Monster {
+  id: number;
+  name: string;
+  type: string;
+  description: string;
+  locations: [];
+}
+
+const $row = document.querySelector('.row');
+
+async function makeMonsters(): Promise<void> {
+  if (!$row) throw new Error('Could not load row');
+  for (let i = 1; i < 59; i++) {
+    try {
+      const response = await fetch(`https://mhw-db.com/monsters/${i}`);
+      if (!response.ok) {
+        throw new Error('This is not ok');
+      }
+      const monster = (await response.json()) as Monster;
+      console.log(monster);
+      const newPage = renderMonster(monster);
+      $row.append(newPage);
+    } catch (error) {
+      console.log('Error:', error);
+    }
+  }
+}
+makeMonsters();
+
+function renderMonster(object: Monster): HTMLDivElement {
+  /* <div class="row">
+      <div class="column-one-third">
+        <img src='link' class="monster-img"/>
+        <div class="monster-name">
+      </div>
+      </div>
+  */
+
+  object.name = object.name.replace(' ', '-');
+  object.name = object.name.replace("'", '-');
+
+  const $oneThird = document.createElement('div');
+  $oneThird.className = 'column-one-third';
+
+  const $monsterList = document.createElement('div');
+  $monsterList.className = 'monster-card';
+
+  const $monImg = document.createElement('img');
+  $monImg.className = 'monster-img';
+  $monImg.setAttribute('src', `images/icons/${object.name}.webp`);
+
+  const $monName = document.createElement('div');
+  $monName.className = 'monster-name';
+
+  const $monsterName = document.createElement('h2');
+  $monsterName.textContent = object.name;
+
```

Comment:
We should change the `textContent` earlier because of the `replace` method you are using. If you wait for here then something like `Xeno'jiiva` will be listed as `Xeno-jiiva`. We just want to do this for the image.